WireDomain/Sources/WireDomain/Notifications/Components/VerifyUserStep.swift

```diff
@@ -0,0 +1,146 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import NeedleFoundation
+import UserNotifications
+import WireAPI
+import WireCrypto
+import WireDataModel
+import WireFoundation
+
+protocol VerifyUserDependency: Dependency {
+    var userID: UUID! { get }
+    var applicationIdentifier: String { get }
+    var applicationContainer: URL { get }
+}
+
+protocol VerifyUserStepFactory {
+    func verifyUserSession() async throws
+}
+
+final class VerifyUserStep: Component<VerifyUserDependency>, VerifyUserStepFactory {
+
+    enum Failure: Error {
+        case noAccountFound
+    }
+
+    public var selectedAccount: Account!
```

Comment:
suggestion: I would avoid this pattern of implicitly unwrapped dependencies. Instead I would suggest to create it in scope, use it in scope, and if it needs to be passed to children, inject it as a dependency to the child component.